ladybug_geometry/geometry3d/polyface.py

```diff
@@ -38,7 +38,8 @@ class Polyface3D(Base2DIn3D):
     __slots__ = ('_vertices', '_faces', '_edges',
                  '_naked_edges', '_internal_edges', '_non_manifold_edges',
                  '_face_indices', '_edge_indices', '_edge_types'
-                 '_min', '_max', '_center', '_area', '_is_solid')
+                 '_min', '_max', '_center', '_area', '_is_solid',
+                 '_tolerance', '_angle_tolerance')
```

Comment:
Should we keep `tolerance` and `angle tolerance` out and as a document properties similar to Rhino? Do we really need to add them to every PolyFace?

Reply:
I see it now how it is used to initiate the PolyFace from dictionary. One solution might be adding default values for them in methods and then use the keys in dictionary as optional values to overwrite them if provided. In that case we don't have to add them as PolyFace properties.

Reply:
Keeping `tolerance` and `angle_tolerance` out of the objects themselves is preferred and it is what I have done on every other object in the library.  I guess the main reason why I could not do it here is because I made `is_solid` a property of the Polyface and used it to auto-run the function that ensures all faces are pointing outward.  Contrarily, for the `Face3D` object, there's no `is_planar` property and there's just a function to `validate_planarity`.  Maybe I should follow the same strategy here and just have a `validate_solidity` method.

I am realizing that what I have here definitely has a code smell to it.  Let me think about it a little and I think a larger change might be needed to ensure that things are a lot cleaner.

Reply:
It took some time but I have finished refactoring the Face3D and Polyface3D `__init__` methods in a way that makes more sense in relation to their `to_dict` output.  Even though it took a bit more of up-front work, I think the code is a lot cleaner now.  The dict format of polyfaces is a bit closer to a mesh now except polyface case can have more than 4 vertices and can possess holes.

Reply:
Also, to go back to the original issue, we no longer store tolerance values on the object.  Everything is re-create-able from the mesh-like data noted above.